elan/rpc/gc.go

```diff
@@ -72,20 +73,16 @@ func (s *server) list(ctx context.Context, prefix, prefix2 string) ([]*ppb.Actio
 
 func (s *server) Delete(ctx context.Context, req *ppb.DeleteRequest) (*ppb.DeleteResponse, error) {
 	log.Notice("Delete request for %d action results, %d blobs", len(req.ActionResults), len(req.Blobs))
-	// Note that we don't differentiate between compressed and uncompressed blobs in the request
-	// (the fact they're stored differently is an implementation detail). This is a _little_
-	// inefficient since we have to check all of them twice, but c'est la vie.
 	var g multierror.Group
-	g.Go(func() error { return s.deleteAll(ctx, "ac", req.ActionResults, req.Hard) })
-	g.Go(func() error { return s.deleteAll(ctx, "cas", req.Blobs, req.Hard) })
-	g.Go(func() error { return s.deleteAll(ctx, "zstd_cas", req.Blobs, req.Hard) })
+	g.Go(func() error { return s.deleteAll(ctx, req.ActionResults, req.Hard) })
+	g.Go(func() error { return s.deleteAll(ctx, req.Blobs, req.Hard) })
```

Comment:
Perhaps the other way of doing this would be to give DeleteRequest  both a `Blobs` and a `CompressedBlobs` field to distinguish them, but maybe that makes things more complicated?

Reply:
That would definitely make things quicker but you are right it would make things a little complicated. I've only ran this in dry_run so don't know how long the delete part actually takes. Maybe a good idea would be to land this, check the times when QAing it, then if time to complete is an issue I can raise a separate revision to speed things up?